world/client.cpp

```diff
@@ -845,83 +849,80 @@ bool Client::HandleEnterWorldPacket(const EQApplicationPacket *app) {
 			}
 		}
 	}
 
-	outapp = new EQApplicationPacket(OP_MOTD);
-	std::string tmp;
-	if (database.GetVariable("MOTD", tmp)) {
-		outapp->size = tmp.length() + 1;
+	auto outapp = new EQApplicationPacket(OP_MOTD);
+	std::string motd_message;
+	if (database.GetVariable("MOTD", motd_message)) {
+		outapp->size = motd_message.length() + 1;
 		outapp->pBuffer = new uchar[outapp->size];
-		memset(outapp->pBuffer,0,outapp->size);
-		strcpy((char*)outapp->pBuffer, tmp.c_str());
-
-	} else {
-		// Null Message of the Day. :)
+		memset(outapp->pBuffer, 0, outapp->size);
+		strcpy((char*)outapp->pBuffer, motd_message.c_str());
+	} else { // Null Message of the Day. :)
 		outapp->size = 1;
 		outapp->pBuffer = new uchar[outapp->size];
 		outapp->pBuffer[0] = 0;
 	}
+
 	QueuePacket(outapp);
 	safe_delete(outapp);
 
 	// set mailkey - used for duration of character session
-	int MailKey = emu_random.Int(1, INT_MAX);
+	int mail_key = emu_random.Int(1, INT_MAX);
 
-	database.SetMailKey(charid, GetIP(), MailKey);
+	database.SetMailKey(charid, GetIP(), mail_key);
 	if (UCSServerAvailable_) {
-		const WorldConfig *Config = WorldConfig::get();
+		auto config = WorldConfig::get();
 		std::string buffer;
 
-		EQ::versions::UCSVersion ConnectionType = EQ::versions::ucsUnknown;
+		auto connection_type = EQ::versions::ucsUnknown;
 
 		// chat server packet
 		switch (GetClientVersion()) {
-		case EQ::versions::ClientVersion::Titanium:
-			ConnectionType = EQ::versions::ucsTitaniumChat;
-			break;
-		case EQ::versions::ClientVersion::SoF:
-			ConnectionType = EQ::versions::ucsSoFCombined;
-			break;
-		case EQ::versions::ClientVersion::SoD:
-			ConnectionType = EQ::versions::ucsSoDCombined;
-			break;
-		case EQ::versions::ClientVersion::UF:
-			ConnectionType = EQ::versions::ucsUFCombined;
-			break;
-		case EQ::versions::ClientVersion::RoF:
-			ConnectionType = EQ::versions::ucsRoFCombined;
-			break;
-		case EQ::versions::ClientVersion::RoF2:
-			ConnectionType = EQ::versions::ucsRoF2Combined;
-			break;
-		default:
-			ConnectionType = EQ::versions::ucsUnknown;
-			break;
+			case EQ::versions::ClientVersion::Titanium:
+				connection_type = EQ::versions::ucsTitaniumChat;
+				break;
+			case EQ::versions::ClientVersion::SoF:
+				connection_type = EQ::versions::ucsSoFCombined;
+				break;
+			case EQ::versions::ClientVersion::SoD:
+				connection_type = EQ::versions::ucsSoDCombined;
+				break;
+			case EQ::versions::ClientVersion::UF:
+				connection_type = EQ::versions::ucsUFCombined;
+				break;
+			case EQ::versions::ClientVersion::RoF:
+				connection_type = EQ::versions::ucsRoFCombined;
+				break;
+			case EQ::versions::ClientVersion::RoF2:
+				connection_type = EQ::versions::ucsRoF2Combined;
+				break;
+			default:
+				connection_type = EQ::versions::ucsUnknown;
+				break;
 		}
 
-		buffer = StringFormat("%s,%i,%s.%s,%c%08X",
-			Config->ChatHost.c_str(),
-			Config->ChatPort,
-			Config->ShortName.c_str(),
+		buffer = fmt::format("{},{},{}.{},{}{}",
```

Comment:
This will break the formatting of the buffer since it was using a char code for connection type and padded hex for the mail key.

It will need to be `fmt::format("{},{},{}.{},{}{:08X}`

The fmt version we use doesn't like using `{:c}` for the connection type formatting for some reason (a newer version worked) so the connection type will need to be explicitly cast to format as a char instead:

`static_cast<char>(connection_type)`

The newer version of fmt doesn't even need the explicit cast so I think it just detects the enum base now

Reply:
Mail keys will definitely need to be tested here, I broke this one myself converting to a string 

Mail is used so infrequently the last thing we want out of this is it being broken for over a year